app/javascript/dashboard/components/widgets/WootWriter/Editor.vue

```diff
@@ -391,6 +406,71 @@ export default {
       tr.scrollIntoView();
       return false;
     },
+    openFileBrowser() {
+      this.$refs.imageUpload.click();
+    },
+    onFileChange() {
+      const file = this.$refs.imageUpload.files[0];
+      if (checkFileSizeLimit(file, MAXIMUM_FILE_UPLOAD_SIZE)) {
+        this.uploadImageToStorage(file);
+      } else {
+        this.showAlert(
+          this.$t(
+            'PROFILE_SETTINGS.FORM.MESSAGE_SIGNATURE_SECTION.IMAGE_UPLOAD_SIZE_ERROR',
+            {
+              size: MAXIMUM_FILE_UPLOAD_SIZE,
+            }
+          )
+        );
+      }
+
+      this.$refs.imageUpload.value = '';
+    },
+    async uploadImageToStorage(file) {
+      try {
+        const { fileUrl } = await uploadFile(file);
+        if (fileUrl) {
+          this.onImageInsertInEditor(fileUrl);
+        }
+        this.showAlert(
+          this.$t(
+            'PROFILE_SETTINGS.FORM.MESSAGE_SIGNATURE_SECTION.IMAGE_UPLOAD_SUCCESS'
+          )
+        );
+      } catch (error) {
+        this.showAlert(
+          this.$t(
+            'PROFILE_SETTINGS.FORM.MESSAGE_SIGNATURE_SECTION.IMAGE_UPLOAD_ERROR'
+          )
+        );
+      }
+    },
+    onImageInsertInEditor(fileUrl) {
+      const { selection, schema, tr } = this.editorView.state;
+      const { nodes } = schema;
+      const currentNode = selection.$from.node();
+      const {
+        type: { name: typeName },
+        content: { size, content },
+      } = currentNode;
+
+      const imageNode = nodes.image.create({ src: fileUrl });
+      if (imageNode) {
+        const isInParagraph = typeName === 'paragraph';
+        const needsNewLine =
+          !content.some(n => n.type.name === 'image') && size !== 0 ? 1 : 0;
+        const nodeToInsert = isInParagraph
+          ? imageNode
+          : nodes.paragraph.create({}, imageNode);
+
+        this.editorView.dispatch(
+          tr
+            .insert(selection.from + needsNewLine, nodeToInsert)
+            .scrollIntoView()
+        );
+        this.focusEditorInputField();
```

Comment:
Can we add a spec or figure out an automated way to test this to make sure that this change is not breaking anything else?

@scmmishra

Reply:
@pranavrajs Done